src/history/hs.c

```diff
@@ -25,6 +25,38 @@ __hs_store_time_pair(WT_SESSION_IMPL *session, wt_timestamp_t timestamp, uint64_
     session->orig_txnid_to_las = txnid;
 }
 
+/*
+ * __hs_start_internal_setup_session --
+ *     Create a temporary internal session to setup history store.
+ */
+static int
+__hs_start_internal_setup_session(WT_SESSION_IMPL *session, WT_SESSION_IMPL **int_sessionp)
+{
+    uint32_t new_session_flags;
+
+    new_session_flags = 0;
+    WT_ASSERT(session, !F_ISSET(session, WT_CONN_HS_OPEN));
+    return (
+      __wt_open_internal_session(S2C(session), "hs_setup", true, new_session_flags, int_sessionp));
+}
+
+/*
+ * __hs_release_internal_setup_session --
+ *     Release the temporary internal session started to setup history store.
+ */
+static int
+__hs_release_internal_setup_session(WT_SESSION_IMPL *session, WT_SESSION_IMPL *int_session)
+{
+    WT_SESSION *wt_session;
+
+    if (session != int_session) {
```

Comment:
Does this condition ever happen in the current code? It looks as though you go to some lengths to *not* call this function if you don't also create a temporary session, so having a check here makes me curious about when it's necessary.

Reply:
I believe I was just being careful that the start/release function calls are always called in pair. I have no strong opinion on either keeping them or removing them as current code will not be breaking the expected semantics.

Reply:
On second though, if I add the NULL value checking for the temporary session in calling function (as you suggested in previous comment ), we can remove this check.